diff --git a/MODULE.bazel b/MODULE.bazel
index b631904..31042c1 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -103,18 +103,12 @@ use_repo(musl, "musl_libc")
 mingw = use_extension("//runtimes/mingw/extension:mingw.bzl", "mingw")
 use_repo(mingw, "mingw")
 
-http_pkg_archive = use_repo_rule("//:http_pkg_archive.bzl", "http_pkg_archive")
-
-http_pkg_archive(
+http_archive(
     name = "macosx15.4.sdk",
+    urls = ["https://github.com/hexops-graveyard/sdk-macos-11.3/archive/ccbaae84cc39469a6792108b24480a4806e09d59.tar.gz"],
+    integrity = "sha256-EYcKSj04K3g0mGEIEmSSG7iDRAp+Cz3UoAc3PYcySjg=",
     build_file = "//third_party/macosx.sdk:BUILD.MacOSX15.4.sdk.tpl",
-    sha256 = "ba3453d62b3d2babf67f3a4a44e8073d6555c85f114856f4390a1f53bd76e24a",
-    strip_files = [
-        "Library/Developer/CommandLineTools/SDKs/MacOSX15.5.sdk/System/Library/Frameworks/Ruby.framework/Versions/Current/Headers/ruby",
-    ],
-    strip_prefix = "Library/Developer/CommandLineTools/SDKs/MacOSX15.5.sdk",
-    # urls = ["https://swcdn.apple.com/content/downloads/10/32/082-12052-A_AHPGDY76PT/1a419zaf3vh8o9t3c0usblyr8eystpnsh5/CLTools_macOSNMOS_SDK.pkg"],
-    urls = ["https://swcdn.apple.com/content/downloads/52/01/082-41241-A_0747ZN8FHV/dectd075r63pppkkzsb75qk61s0lfee22j/CLTools_macOSNMOS_SDK.pkg"],
+    strip_prefix = "sdk-macos-11.3-ccbaae84cc39469a6792108b24480a4806e09d59/root",
 )
 
 glibc = use_extension("//runtimes/glibc/extension:glibc.bzl", "glibc")
