diff --git a/cc/toolchains/args/archiver_flags/BUILD b/cc/toolchains/args/archiver_flags/BUILD
index 1d4f6c5..860ef16 100644
--- a/cc/toolchains/args/archiver_flags/BUILD
+++ b/cc/toolchains/args/archiver_flags/BUILD
@@ -14,6 +14,7 @@ config_setting(
     name = "use_libtool_on_macos_setting",
     constraint_values = ["@platforms//os:macos"],
     flag_values = {":use_libtool_on_macos": "true"},
+    visibility = ["//visibility:public"],
 )
 
 cc_feature(
